### PR TITLE
feat: Implements NullType and NullVector

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR implements NullType and NullVector for the datatypes2 crate.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #555

### CODE REVIEW

File: src/datatypes2/src/types/null_type.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    fn default_value(&self) -> Value {
+        Value::Null
+    }
+
+    fn as_arrow_type(&self) -> ArrowDataType {
+        ArrowDataType::Null
+    }
+
+    fn create_mutable_vector(&self, _capacity: usize) -> Box<dyn MutableVector> {
+        Box::new(NullVectorBuilder::default())
```
Comment (waynexia) [inline src/datatypes2/src/types/null_type.rs:52]:
```suggestion
    fn create_mutable_vector(&self, capacity: usize) -> Box<dyn MutableVector> {
        Box::new(NullVectorBuilder::with_capacity(capacity))
```

**Review by waynexia — APPROVED** [review]

File: src/datatypes2/src/vectors/operations/find_unique.rs [hunk 2]
```diff
@@ -60,20 +61,20 @@ pub(crate) fn find_unique_scalar<'a, T: ScalarVector>(
...
+    if vector.is_empty() {
+        return;
+    }
 
-//     let is_first_not_duplicate = prev_vector.map(|pv| pv.is_empty()).unwrap_or(true);
-//     if is_first_not_duplicate {
-//         selected.set(0, true);
-//     }
-// }
+    let is_first_not_duplicate = prev_vector.map(|pv| pv.is_empty()).unwrap_or(true);
```
Comment (v0y4g3r) [inline src/datatypes2/src/vectors/operations/find_unique.rs:73]:
```suggestion
    let is_first_not_duplicate = prev_vector.map(NullVector::is_empty).unwrap_or(true);
```

File: src/datatypes2/src/vectors/operations/find_unique.rs [hunk 3]
```diff
@@ -60,20 +61,20 @@ pub(crate) fn find_unique_scalar<'a, T: ScalarVector>(
...
-//     if vector.is_empty() {
-//         return;
-//     }
+pub(crate) fn find_unique_null(
+    vector: &NullVector,
+    selected: &mut BitVec,
+    prev_vector: Option<&NullVector>,
+) {
+    if vector.is_empty() {
+        return;
```
Comment (v0y4g3r) [inline src/datatypes2/src/vectors/operations/find_unique.rs:70]: Should we set all position of `selected` to `false` here ?

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/datatypes2/src/vectors/operations/find_unique.rs (hunk 3 above)
Comment (evenyag) [reply src/datatypes2/src/vectors/operations/find_unique.rs:70]: selected should also be empty

**Pull request merged**